crates/aesgcm/src/core/aesgcm.rs

```diff
@@ -1,78 +1,119 @@
+/// reused with the same key. The length of the `nonce` is determined by the
+/// `Aes256GcmRust::NONCE_LENGTH` constant,
+/// * `plaintext`: The `plaintext` parameter is the data that you want to
+///   encrypt.
+/// It is a slice of bytes (`&[u8]`) that represents the raw data that you want
+/// to protect.
+/// * `authenticated_data`: The `authenticated_data` parameter is additional
+///   data
+/// that is authenticated but not encrypted. It is used to provide integrity and
+/// authenticity to the encrypted data. This data is included in the
+/// authentication tag calculation but is not included in the encrypted output.
 ///
-/// * `nonce`: The `nonce` property is a 12-byte array used as a unique value
-///   for each encryption operation in the AES-GCM encryption mode. It is
-///   important that the nonce is never reused with the same key, as this can
-///   compromise the security of the encryption.
-pub struct ReExposedAesGcm {
-    cipher: Aes256Gcm,
-    nonce: GenericArray<u8, U12>,
+/// Returns:
+///
+/// a `Result` type, specifically `Result<Vec<u8>, AesGcmError>`. This means
+/// that the function can either return an `Ok` variant containing a `Vec<u8>`
+/// (the encrypted data) or an `Err` variant containing an `AesGcmError` (an
+/// error occurred during encryption).
+pub fn encrypt(
+    key: [u8; Aes256GcmRust::KEY_LENGTH],
+    nonce: [u8; Aes256GcmRust::NONCE_LENGTH],
+    plaintext: &[u8],
+    authenticated_data: &[u8],
+) -> Result<Vec<u8>, AesGcmError> {
+    let key = SymmetricKey::try_from_bytes(key)?;
+    let nonce = Nonce::try_from_bytes(nonce)?;
+    Ok(Aes256GcmRust::new(&key).encrypt(&nonce, plaintext, Some(authenticated_data))?)
 }
 
-impl ReExposedAesGcm {
-    /// This function instantiates an AES-256-GCM cipher object with a given key
-    /// and nonce.
-    ///
-    /// Arguments:
-    ///
-    /// * `key`: The `key` parameter is a reference to a byte array of length
-    ///   32, which is used to create an instance of the `Aes256Gcm` cipher
-    ///   object. This key is used to encrypt and decrypt data using the cipher.
-    /// * `nonce`: The `nonce` parameter is a 12-byte array representing a
-    ///   unique value used in the encryption process to ensure that each
-    ///   message encrypted with the same key is unique. It stands for "number
-    ///   used once".
-    pub fn instantiate(
-        key: &[u8; KEY_LENGTH],
-        nonce: &[u8; NONCE_LENGTH],
-    ) -> Result<Self, AesGcmError> {
-        // Create the cipher object
-        let cipher = Aes256Gcm::new_from_slice(key)?;
-
-        // Transformed from a byte array:
-        let nonce = *Nonce::from_slice(nonce);
-
-        Ok(Self { cipher, nonce })
-    }
-
-    pub fn encrypt(&self, plaintext: &[u8]) -> Result<Vec<u8>, AesGcmError> {
-        Ok(self.cipher.encrypt(&self.nonce, plaintext)?)
+/// The `decrypt` function takes a key, nonce, ciphertext, and authenticated
```

Comment:
Same: please shorten this doc. No one will ever read a 30-lines long documentation. Even more if this is to say "This function decrypts the given AES-256 GCM ciphertext using the provided key, nonce and authentication data".

Reply:
done